Translate useDeferredValue.md (#826)
reactjs/es.react.dev: src/content/reference/react/useDeferredValue.md

<Intro>

> `useDeferredValue` es un Hook de React que permite realizar una actualización en diferido de una parte de la interfaz de usuario (_UI_).

Reviewer: ```suggestion
`useDeferredValue` es un Hook de React que permite realizar una actualización en diferido de una parte de la UI.
```

---

> ## Referencias {/*reference*/}

Reviewer: ```suggestion
## Referencia {/*reference*/}
```

}
```

[Ver más ejemplos abajo.](#usage)

#### Parámetros {/*parameters*/}

* `value`: El valor que se quiere diferir. Puede ser de cualquier tipo.

> #### Retornos {/*returns*/}

Reviewer: ```suggestion
#### Devuelve {/*returns*/}
```

Durante el renderizado inicial, el valor diferido devuelto será el mismo que el valor que se haya proporcionado inicialmente. Durante las actualizaciones, React realizará un primer intento de re-renderizado con el valor anterior (de modo que devolverá el valor anterior) e intentará realizar otro re-renderizado en segundo plano con el nuevo valor (por lo que devolverá el valor actualizado).

> #### Limitaciones {/*caveats*/}

Reviewer: ```suggestion
#### Advertencias {/*caveats*/}
```

> - Los valores que se pasen a `useDeferredValue` deben ser tanto valores primitivos (como `string` y `number`) u objetos creados fuera del proceso de renderización. Si se crea un nuevo objeto durante el el proceso de renderización e inmediatamente se le pasa a `useDeferredValue` generará un valor distinto en cada renderizado causando re-renderizados innecesarios en segundo plano.

Reviewer: ```suggestion
- Los valores que pases a `useDeferredValue` deben ser tanto valores primitivos (como `string` y `number`) u objetos creados fuera del proceso de renderización. Si se crea un nuevo objeto durante el el proceso de renderización e inmediatamente se le pasa a `useDeferredValue` generará un valor distinto en cada renderizado causando re-renderizados innecesarios en segundo plano.
```

> - Cuando `useDeferredValue` reciba un valor distinto (realizado una comparación mediante [`Object.is`](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Object/is)), además del renderizado actual (cuando todavía se está empleando el valor anterior), se programa un re-renderizado en segundo plano con el nuevo valor. Este re-renderizado en segundo plano se puede interrumpir: Si se produce otra actualización al `valor`, React comenzará desde cero el re-renderizado en segundo plano. Por ejemplo, si el usuario está escribiendo en un `input` mas rápido de lo que un gráfico es capaz de recibir su valor diferido para renderizarse de nuevo, el gráfico solo se volverá a renderizar después de que el usuario deje de escribir.

Reviewer: ```suggestion
- Cuando `useDeferredValue` recibe un valor diferente (comparado con [`Object.is`](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Object/is)), además de la renderización actual (cuando aún utiliza el valor anterior), programa una re-renderización en segundo plano con el nuevo valor. La renderización en segundo plano es interrumpible: si hay otra actualización del `valor`, React reiniciará la renderización en segundo plano desde cero. Por ejemplo, si el usuario está escribiendo en una entrada más rápido de lo que un gráfico que recibe su valor diferido puede volver a renderizar, el gráfico sólo se volverá a renderizar cuando el usuario deje de escribir.
```

> - Cuando `useDeferredValue` reciba un valor distinto (realizado una comparación mediante [`Object.is`](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Object/is)), además del renderizado actual (cuando todavía se está empleando el valor anterior), se programa un re-renderizado en segundo plano con el nuevo valor. Este re-renderizado en segundo plano se puede interrumpir: Si se produce otra actualización al `valor`, React comenzará desde cero el re-renderizado en segundo plano. Por ejemplo, si el usuario está escribiendo en un `input` mas rápido de lo que un gráfico es capaz de recibir su valor diferido para renderizarse de nuevo, el gráfico solo se volverá a renderizar después de que el usuario deje de escribir.

Reviewer: ```suggestion
- Cuando `useDeferredValue` recibe un valor diferente (comparado con [`Object.is`](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Object/is)), además de la renderización actual (cuando aún utiliza el valor anterior), programa una re-renderización en segundo plano con el nuevo valor. La renderización en segundo plano es interrumpible: si hay otra actualización del `valor`, React reiniciará la renderización en segundo plano desde cero. Por ejemplo, si el usuario está escribiendo en un `input` más rápido de lo que un gráfico que recibe su valor diferido puede volver a renderizar, el gráfico solo se volverá a renderizar cuando el usuario deje de escribir.
```

> - `useDeferredValue` está integrado con [`<Suspense>`.](/reference/react/Suspense). Si la actualización en segundo plano causada por un nuevo valor suspende la interfaz de usuario (_UI_), el usuario no podrá ver el _fallback_. En su lugar, podrá el valor diferido anterior hasta que se carguen los datos.

Reviewer: ```suggestion
- `useDeferredValue` está integrado con [`<Suspense>`.](/reference/react/Suspense) Si la actualización en segundo plano causada por un nuevo valor suspende la UI, el usuario no verá el `fallback`. Verá el antiguo valor diferido hasta que se carguen los datos.
```

> - `useDeferredValue` no previene peticiones de red por sí mismo.

Reviewer: ```suggestion
- `useDeferredValue` no previene por sí mismo las peticiones de red adicionales.
```

- No existe un retardo fijo causado por `useDeferredValue`. Tan pronto como React finalice el renderizado original, inmediatamente, empezará a trabajar sobre el re-renderizado en segundo plano con el nuevo valor diferido. Cualquier actualización causada por eventos (como escribir por teclado) interrumpirá y tendrá prioridad respecto al proceso de re-renderizado en segundo plano.

> - El re-renderizado en segundo plano causado por `useDeferredValue` no dispara Efectos hasta que se haya confirmado en pantalla. Si el proceso de re-renderizado en segundo plano se suspende, los Efectos volverán a ejecutarse una vez los datos hayan sido cargados y la interfaz de usuario (_UI_) se haya actualizado.

Reviewer: ```suggestion
- El re-renderizado en segundo plano causado por `useDeferredValue` no dispara Efectos hasta que se haya confirmado en pantalla. Si el proceso de re-renderizado en segundo plano se suspende, los Efectos volverán a ejecutarse una vez los datos hayan sido cargados y la UI se haya actualizado.
```

---

> ## Cómo utilizar {/*usage*/}

Reviewer: ```suggestion
## Uso {/*usage*/}
```

> ### Mostrando contenido desactualizado mientras se carga el contenido actualizado. {/*showing-stale-content-while-fresh-content-is-loading*/}

Reviewer: ```suggestion
### Mostrar contenido desactualizado mientras se carga el contenido actualizado. {/*showing-stale-content-while-fresh-content-is-loading*/}
```

> Llama a `useDeferredValue` en el nivel superior de tu componente para retrasar la actualización de alguna parte de tu interfaz de usuario (_UI_).

Reviewer: ```suggestion
Llama a `useDeferredValue` en el nivel superior de tu componente para retrasar la actualización de alguna parte de tu UI.
```

</Sandpack>

> Una alternativa común en la interfaz de usuario (_UI_) es *diferir* la actualización de las listas de resultados y seguir mostrando los anteriores resultados hasta que los nuevos estén disponibles. Llama a `useDeferredValue` para pasar una versión diferida de la `query`:

Reviewer: ```suggestion
Una alternativa común en la UI es *diferir* la actualización de las listas de resultados y seguir mostrando los anteriores resultados hasta que los nuevos estén disponibles. Llama a `useDeferredValue` para pasar una versión diferida de la `query`:
```

<DeepDive>

#### ¿Cómo funciona realmente diferir un valor? {/*how-does-deferring-a-value-work-under-the-hood*/}

Puedes pensar que ocurre de acuerdo a estos dos pasos:

1. **En primer lugar, React re-renderiza con la nueva `query` (`"ab"`) pero utilizando el anterior `deferredQuery` (cuyo valor aún es `"a")`.** El valor de `deferredQuery`, el cual se pasa a la lista resultante, está "diferido" respecto al valor de la `query`.

> 2. **En segundo plano, React intentará re-renderizar con *ambos* `query` y `deferredQuery` actualizados con el valor `"ab"`.** Si este re-renderizado se completa, React lo mostrará por pantalla. De lo contrario, si se "suspende" (los resultados pra `"ab"` aún no se han cargado), React abandonará este intento de renderización y re-intentará este re-renderizado nuevamente una vez los datos hayan sido cargados. El usuario seguirá viendo el valor diferido obsoleto hasta que los datos hayan sido cargados.

Reviewer: ```suggestion
2. **En segundo plano, React intentará re-renderizar con *ambos* `query` y `deferredQuery` actualizados con el valor `"ab"`.** Si este re-renderizado se completa, React lo mostrará por pantalla. De lo contrario, si se "suspende" (los resultados para `"ab"` aún no se han cargado), React abandonará este intento de renderización y re-intentará este re-renderizado nuevamente una vez los datos hayan sido cargados. El usuario seguirá viendo el valor diferido obsoleto hasta que los datos hayan sido cargados.
```

La renderización diferida en segundo plano se puede interrumpir. Por ejemplo, si escribimos en el _input_ nuevamente, React abandonará esa renderización y comenzará una nueva con el nuevo valor. React siempre utilizará el último valor proporcionado.

> Ten en cuenta que aún se realizará una nueva petición de red por cada vez que se actualice el valor del _input_ por teclado nuevamente. Lo que se "difiere" realmente es la visualización de los resultados (hasta que estén disponibles) y no las peticiones de red. Los valores introducidos por el usuario son cacheados, por lo que modificar el valor actual a uno utilizado anteriormente mostrará los resultados, almacenados en la cache, inmediatamente y no se realizará una nueva petición de red.

Reviewer: ```suggestion
Ten en cuenta que sigue habiendo una petición de red por cada pulsación de tecla. Lo que se aplaza aquí es la visualización de los resultados (hasta que estén listos), no las peticiones de red en sí. Incluso si el usuario continúa escribiendo, las respuestas para cada pulsación de tecla se almacenan en caché, por lo que pulsar _Backspace_ es instantáneo y no se obtiene de nuevo.
```

</DeepDive>

---

> ### Indicando que el contenido es obsoleto {/*indicating-that-the-content-is-stale*/}

Reviewer: ```suggestion
### Indicar que el contenido es obsoleto {/*indicating-that-the-content-is-stale*/}
```

---

> ### Diferir el re-renderizando una parte de la interfaz de usuario (_UI_) {/*deferring-re-rendering-for-a-part-of-the-ui*/}

Reviewer: ```suggestion
### Diferir el re-renderizando una parte de la UI {/*deferring-re-rendering-for-a-part-of-the-ui*/}
```

> Puedes utilizar `useDeferredValue` como medio para optimizar el rendimiento. Es útil cuando una parte de tu interfaz de usuario (_UI_) es más lenta a la hora de re-renderizar y no existe una forma fácil de optimizarlo a fin de evitar que otras partes de la interfaz de usuario (_UI_) se bloqueen.

Reviewer: ```suggestion
Puedes utilizar `useDeferredValue` como medio para optimizar el rendimiento. Es útil cuando una parte de tu UI es más lenta a la hora de re-renderizar y no existe una forma fácil de optimizarlo a fin de evitar que otras partes de la UI se bloqueen.
```

> Imagina un _input_ y un componente (como un gráfico o una lista con muchos datos) que se re-renderiza cada vez que se introduce un nuevo valor en el _input_:

Reviewer: ```suggestion
Imagina que tienes un campo de texto _input_ y un componente (como un gráfico o una lista larga) que se vuelve a mostrar cada vez que pulsas una tecla:
```
